### ISSUE TITLE
Invalid path issue

### ISSUE THREAD

**Issue by pradp** [cloudfoundry/cli#744]
I'm getting the below error even with the valid path for the file. I see this issue after updating to the latest version.

```
Machine - Windows 7 64-bit
cf version 6.15.0+fa1bfe2-2016-01-12
```

```
cf push redis-example -p C:\Users\sekhap\Desktop\pcf-spring-redis-example-0.0.1-SNAPSHOT.jar --no-start
```

```
RESPONSE: [2016-01-28T19:41:35+05:30]
HTTP/1.1 400 Bad Request
Content-Length: 117
Content-Type: application/json;charset=utf-8
Date: Thu, 28 Jan 2016 14:11:35 GMT
Server: nginx
X-Cf-Requestid: 6aaf5f04-451d-4cbb-5945-68d6329592dc
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: 10eb8f59-8bc1-4c28-531c-767150dc1d93::3200de4c-a073-4706-9e96-00971b81c1f7

{
  "code": 10005,
  "description": "The query parameter is invalid: path",
  "error_code": "CF-BadQueryParameter"
}

FAILED
Server error, status code: 400, error code: 10005, message: The query parameter is invalid: path
FAILED
Server error, status code: 400, error code: 10005, message: The query parameter is invalid: path
```

**Comment by dkoper** [comment]
Hi @pradp,

Can you let us know the API version returned by `cf api`?
Also, can you run `cf curl /v2/info` and let us know the value of the "api_version" field, if different from the one returned above?

Cheers,
Dies Koper
CF CLI PM

**Comment by pradp** [comment]
Hi @dkoper 

I tried today the same command and it is working now. I'm working with private cloud set up and yesterday there was an upgrade to PCF 1.6 version. May be this could have fixed the issue but I'm curious on the API version. If you look at the below details still there is still a mismatch but it works. Can you explain a bit on this? Thank you.

```
λ cf api
API endpoint: https://api.xxx.xxx.xxx.com (API version: 2.23.0)

λ cf curl /v2/info
{
   "name": "vcap",
   "build": "2222",
   "support": "https://support.pivotal.io",
   "version": 2,
   "description": "Cloud Foundry sponsored by Pivotal",
   .....
   "min_cli_version": "6.7.0",
   "min_recommended_cli_version": "6.11.2",
   "api_version": "2.43.0",
   .....
}
```

**Comment by dkoper** [comment]
Hi @pradp,

`cf api` returns a cached API version, from the last time you logged in. That can be confusing, as reported in #558. `cf curl` makes an actual HTTP request to get the current version.

API version 2.43.0 supports the route with path feature. When specifying the --route-path option, the CLI checks the (cached) API version and fails with a meaningful error message if its version doesn't support this feature.
I am concerned that what you saw is the CLI including the `path` field in a request to an older API endpoint even if you haven't specified the option. I have enough information to test that locally.

Cheers,
Dies Koper
CF CLI PM

**Comment by dkoper** [comment]
Hi @pradp ,

I was unable to reproduce the issue here. Can you provide more of the trace? In particular, I'd like to see the request that was made that caused the Cloud Controller to return that error.

Cheers,
Dies Koper
CF CLI PM

**Comment by pradp** [comment]
Hi @dkoper ,

I too couldn't reproduce the issue myself. I should have taken the request trace before but I missed it. If this issue has occurred because of the older PCF 1.4.2 then I don't have that environment to check it now. Sorry. Is there any way I could help you in this?

Pradeep

**Comment by dkoper** [comment]
Hi Pradeep,

Thanks for checking again.
As the problem is resolved for you, I think I'll close this issue. If someone runs into the issue again and can't upgrade CF, I'll ask for the trace and investigate further then.

Cheers,
Dies Koper
CF CLI PM

**Comment by jhiemer** [comment]
@dkoper here you go:

```
REQUEST: [2016-02-16T11:45:08+01:00]
GET /v2/routes?inline-relations-depth=1&q=host%3Aint-rpt%3Bdomain_guid%3A25aa6dd4-28d6-40b1-b48e-cd9cd06493cd%3Bpath%3A HTTP/1.1
Host: --------
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: application/json
User-Agent: go-cli 6.15.0+fa1bfe2 / darwin



RESPONSE: [2016-02-16T11:45:08+01:00]
HTTP/1.1 400 Bad Request
Content-Length: 104
Connection: keep-alive
Content-Type: application/json;charset=utf-8
Date: Tue, 16 Feb 2016 10:45:08 GMT
Keep-Alive: timeout=20
Server: nginx/1.1.19
X-Cf-Requestid: 79b4a687-dfcc-47e4-63e6-f19b27032178
X-Content-Type-Options: nosniff
X-Vcap-Request-Id: 62c5b9c1-e835-4d43-67d4-17f9559bfc6e::af5c2289-b01b-45d0-b039-f7b150df1b23

{"code":10005,"description":"The query parameter is invalid: path","error_code":"CF-BadQueryParameter"}
```

Curl:
````
cf curl /v2/info 
{
   "build": "189",
   "version": 2,
   "api_version": "2.14.0",
  ....
}
````

Removed the hostnames etc. manually. No deployment possible atm.

**Comment by dkoper** [comment]
This has been fixed in cf CLI v6.16.0.
https://github.com/cloudfoundry/cli#downloads

Regards,
Dies Koper
CF CLI PM
